bridge/client/app/_views/ktb-environment-view/ktb-stage-details/ktb-stage-details.component.ts

```diff
@@ -1,61 +1,48 @@
-import { Subject } from 'rxjs';
-import { Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { ISelectedStageInfo } from '../ktb-environment-view.component';
 
 export type ServiceFilterType = 'evaluation' | 'problem' | 'approval' | undefined;
 
 @Component({
   selector: 'ktb-stage-details',
   templateUrl: './ktb-stage-details.component.html',
   styleUrls: ['./ktb-stage-details.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class KtbStageDetailsComponent implements OnInit, OnDestroy {
-  public _project?: Project;
-  public selectedStage?: Stage;
+export class KtbStageDetailsComponent {
   public filterEventType: ServiceFilterType;
   public overlayConfig: DtOverlayConfig = {
     pinnable: true,
   };
-  public isQualityGatesOnly = false;
+  public isQualityGatesOnly$: Observable<boolean>;
   public filteredServices: string[] = [];
-  private readonly unsubscribe$ = new Subject<void>();
+  private _selectedStageInfo?: ISelectedStageInfo;
 
-  @ViewChild('problemFilterEventButton') public problemFilterEventButton?: DtToggleButtonItem<string>;
-  @ViewChild('evaluationFilterEventButton') public evaluationFilterEventButton?: DtToggleButtonItem<string>;
-  @ViewChild('approvalFilterEventButton') public approvalFilterEventButton?: DtToggleButtonItem<string>;
-
-  @Input()
-  get project(): Project | undefined {
-    return this._project;
-  }
-
-  set project(project: Project | undefined) {
-    if (this._project !== project) {
-      this._project = project;
-      this.selectedStage = undefined;
+  @Input() project?: Project;
+  @Input() set selectedStageInfo(stageInfo: ISelectedStageInfo | undefined) {
+    this._selectedStageInfo = stageInfo;
+    if (stageInfo && this.filterEventType !== stageInfo.filterType) {
+      this.resetFilter(stageInfo.filterType);
     }
   }
-
-  constructor(private dataService: DataService, private router: Router) {}
-
-  ngOnInit(): void {
-    this.dataService.isQualityGatesOnly.pipe(takeUntil(this.unsubscribe$)).subscribe((isQualityGatesOnly) => {
-      this.isQualityGatesOnly = isQualityGatesOnly;
-    });
+  get selectedStageInfo(): ISelectedStageInfo | undefined {
+    return this._selectedStageInfo;
   }
+  @Output() selectedStageInfoChange = new EventEmitter<ISelectedStageInfo>();
 
-  selectStage($event: { stage: Stage; filterType: ServiceFilterType }): void {
-    this.selectedStage = $event.stage;
-    if (this.filterEventType !== $event.filterType) {
-      this.resetFilter($event.filterType);
-    }
+  @ViewChild('problemFilterEventButton') public problemFilterEventButton?: DtToggleButtonItem<string>;
+  @ViewChild('evaluationFilterEventButton') public evaluationFilterEventButton?: DtToggleButtonItem<string>;
+  @ViewChild('approvalFilterEventButton') public approvalFilterEventButton?: DtToggleButtonItem<string>;
+
+  constructor(private dataService: DataService) {
+    this.isQualityGatesOnly$ = this.dataService.isQualityGatesOnly;
```

Comment:
Let us provide an input parameter `isQualityGatesOnly` so that we can get rid of the DataService in this component.